dev/tools/vitool/lib/vitool.dart

```diff
@@ -0,0 +1,560 @@
+      sb.write('${kIndent * 5}],\n');
+    }
+    sb.write('${kIndent * 4}),\n');
+    return sb.toString();
+  }
+}
+
+/// Interprets some subset of an SVG file.
+///
+/// Recursively goes over the SVG tree, applying transforms and opacities,
+/// and build a FrameData which is a flat representation of the paths in the SVG
+/// file, after applying transformations and converting relative coordinates to
+/// absolute.
+///
+/// This does not support the SVG specification, but is just built to
+/// support SVG files exported by a specific tool the motion design team is
+/// using.
+FrameData interpretSvg(String svgFilePath) {
+  final File file = new File(svgFilePath);
+  final String fileData = file.readAsStringSync();
+  final XmlElement svgElement = _extractSvgElement(xml.parse(fileData));
+  final double width = parsePixels(_extractAttr(svgElement, 'width')).toDouble();
+  final double height = parsePixels(_extractAttr(svgElement, 'height')).toDouble();
+
+  final List<SvgPath> paths =
+    _interpretSvgGroup(svgElement.children, new _Transform());
+  return new FrameData(new Point<double>(width, height), paths);
+}
+
+List<SvgPath> _interpretSvgGroup(List<XmlNode> children, _Transform transform) {
+  final List<SvgPath> paths = <SvgPath>[];
+  for (XmlNode node in children) {
+    if (node.nodeType != XmlNodeType.ELEMENT)
+      continue;
+    final XmlElement element = node;
+
+    if (element.name.local == 'path') {
+      paths.add(SvgPath.fromElement(element).applyTransform(transform));
+    }
+
+    if (element.name.local == 'g') {
+      double opacity = transform.opacity;
+      if (_hasAttr(element, 'opacity'))
+        opacity *= double.parse(_extractAttr(element, 'opacity'));
+
+      Matrix3 transformMatrix = transform.transformMatrix;
+      if (_hasAttr(element, 'transform'))
+        transformMatrix = transformMatrix.multiplied(
+          _parseSvgTransform(_extractAttr(element, 'transform')));
+
+      final _Transform subtreeTransform = new _Transform(
+        transformMatrix: transformMatrix,
+        opacity: opacity
+      );
+      paths.addAll(_interpretSvgGroup(element.children, subtreeTransform));
+    }
+  }
+  return paths;
+}
+
+// Given a points list in the form e.g: "25.0, 1.0 12.0, 12.0 23.0, 9.0" matches
+// the coordinated of the first point and the rest of the string, for the
+// example above:
+// group 1 will match "25.0"
+// group 2 will match "1.0"
+// group 3 will match "12.0, 12.0 23.0, 9.0"
+//
+// Commas are optional.
+final RegExp _pointMatcher = new RegExp(r' *([\-\.0-9]+) *,? *([\-\.0-9]+)(.*)');
+
+/// Parse a string with a list of points, e.g:
+/// '25.0, 1.0 12.0, 12.0 23.0, 9.0' will be parsed to:
+/// [Point(25.0, 1.0), Point(12.0, 12.0), Point(23.0, 9.0)].
+///
+/// Commas are optional.
+List<Point<double>> parsePoints(String points) {
+  String unParsed = points;
+  final List<Point<double>> result = <Point<double>>[];
+  while(unParsed.isNotEmpty && _pointMatcher.hasMatch(unParsed)) {
```

Comment:
rather than hasMatch, maybe use startsWith?

Reply:
that way you'll catch cases where the unparsed string starts with bogus text

Reply:
Added ^ to the start of the regex